src/ngraph_builder.cc

```diff
@@ -4187,14 +4187,71 @@ static Status TranslateSelectOp(
   TF_RETURN_IF_ERROR(
       GetInputNodes(ng_op_map, op, &ng_input1, &ng_input2, &ng_input3));
 
-  // broadcast to make all tensors same shape, as required by ngraph select op
-  std::tie(ng_input1, ng_input2) =
-      ng::builder::numpy_broadcast(std::make_pair(ng_input1, ng_input2));
-  std::tie(ng_input2, ng_input3) =
-      ng::builder::numpy_broadcast(std::make_pair(ng_input2, ng_input3));
+  auto ng_input1_size = ng_input1->get_shape().size();
+  auto ng_input2_size = ng_input2->get_shape().size();
+
+  int length;
+  shared_ptr<ng::Node> first, second, ng_input_new;
+  shared_ptr<ng::Node> ng_select;
+  ng::AxisVector ng_axis_order;
+
+  if (ng_input1_size < ng_input2_size) {
+    first = ng_input2;
+    second = ng_input1;
+  } else {
+    first = ng_input1;
+    second = ng_input2;
+  }
+
+  auto first_shape = first->get_shape();
+  auto second_shape = second->get_shape();
+
+  auto first_size = first_shape.size();
+  auto second_size = second_shape.size();
+
+  for (size_t i = 0; i < first_size; i++) {
+    if (second_shape[0] == first_shape[i]) {
+      length = first_size - i - second_size;
+    }
+  }
+  if (length != 0) {
+    std::vector<size_t> tmp_vector((length + second_size), 1);
+    tmp_vector[0] = first_shape[0];
+    for (size_t i = 0; i < second_size; i++) {
+      ng_axis_order.push_back(i);
+    }
+    ng_input_new = ConstructNgNode<ng::op::Reshape>(op->name(), second,
+                                                    ng_axis_order, tmp_vector);
+
+    // broadcast to make all tensors same shape, as required by ngraph select op
+    if (first == ng_input1) {
+      std::tie(ng_input_new, second) =
+          ng::builder::numpy_broadcast(std::make_pair(ng_input_new, second));
+      std::tie(second, ng_input3) =
+          ng::builder::numpy_broadcast(std::make_pair(second, ng_input3));
+      ng_select = ConstructNgNode<ng::op::Select>(op->name(), ng_input_new,
+                                                  second, ng_input3);
+    }
+    if (first == ng_input2) {
+      std::tie(first, ng_input_new) =
+          ng::builder::numpy_broadcast(std::make_pair(first, ng_input_new));
+      std::tie(ng_input_new, ng_input3) =
+          ng::builder::numpy_broadcast(std::make_pair(ng_input_new, ng_input3));
+      ng_select = ConstructNgNode<ng::op::Select>(op->name(), first,
+                                                  ng_input_new, ng_input3);
+    }
```

Comment:
lines 4227-4230 could possibly be done without any if-else I think. but we need to check

For example:
```if (first == ng_input1)
   {ng::builder::numpy_broadcast(std::make_pair(ng_input_new, second));}```
means: `{ng::builder::numpy_broadcast(std::make_pair(ng_input_new, ng_input2));}`
vs
```if (first == ng_input2) 
{ng::builder::numpy_broadcast(std::make_pair(first, ng_input_new));}```
means: `{ng::builder::numpy_broadcast(std::make_pair(ng_input2, ng_input_new));}`

so we are doing a broadcast on `ng_input_new` and `ng_input2` in both cases (except in a different order). After the reshape would the order matter?

eg: are these 2 same:  `numpy_broadcast([5 7 3 2 1], [7 1 1 1])` vs  `numpy_broadcast([7 1 1 1], [5 7 3 2 1])`
I think they are same (but need to check), hence the order does not matter. hence that whole block of if-else is not needed perhaps

Reply:
Figured I do not have to have 2 conditions. Condition tensor and input tensor can either have same shape or when input tensor rank is higher than condition, condition tensor can only be rank 1 as per TF where Op. made changes accordingly.